cmd/skupper/skupper_token.go

```diff
@@ -44,3 +51,68 @@ func NewCmdTokenCreate(skupperClient SkupperTokenClient, flag string) *cobra.Com
 	skupperClient.CreateFlags(cmd)
 	return cmd
 }
+
+var TokenStatusOpts types.TokenStatus
+
+func NewCmdTokenStatus(skupperClient SkupperTokenClient, flag string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "status <output-token-file>",
+		Short:  "display token information",
+		Args:   cobra.ExactArgs(1),
+		PreRun: skupperClient.NewClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			silenceCobra(cmd)
+
+			ctx, cancel := context.WithTimeout(context.Background(), types.DefaultTimeoutDuration)
+			defer cancel()
+
+			currentSiteId, err := skupperClient.GetCurrentSiteId(ctx)
+			if err != nil && strings.HasPrefix(err.Error(), "Skupper is not enabled") {
+				return fmt.Errorf("%s", err.Error())
+			}
+			// loading secret from file
+			yaml, err := os.ReadFile(args[0])
+			if err != nil || yaml == nil {
+				return fmt.Errorf("could not read connection token %w", err)
+			}
+
+			ys := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme,
+				scheme.Scheme)
+			var secret = &corev1.Secret{}
+			_, _, err = ys.Decode(yaml, nil, secret)
+			if err != nil {
+				return fmt.Errorf("could not parse connection token: %w", err)
+			}
+			TokenStatusOpts.Name = secret.ObjectMeta.Name
+			TokenStatusOpts.Secret = secret
+			if secret.ObjectMeta.Labels["skupper.io/type"] == types.TypeClaimRequest {
+				TokenStatusOpts.Type = types.TokenRoleClaim
+			} else {
+				TokenStatusOpts.Type = types.TokenRoleCert
+			}
+			TokenStatusOpts.Creator = types.TokenCreatorRemote
+			if currentSiteId == secret.ObjectMeta.Annotations[types.TokenGeneratedBy] {
+				TokenStatusOpts.Creator = types.TokenCreatorLocal
+			}
+
+			fmt.Printf("Token Name: %s\n", TokenStatusOpts.Name)
+			fmt.Printf("  path: %s\n", args[0])
+			fmt.Printf("  token was generated %s by: %s\n", TokenStatusOpts.Creator, secret.ObjectMeta.Annotations[types.TokenGeneratedBy])
+			fmt.Printf("  token type %s\n", TokenStatusOpts.Type)
+			fmt.Printf("  version: %s\n", secret.TypeMeta.APIVersion)
```

Comment:
I don't think the API version of the Secret resource is of much interest. I would use version as the site version.